### ISSUE TITLE
🐛 Bug Report: (1.32.0) unable to open scaffolder tasks list

### ISSUE THREAD

**Issue by stephenglass** [backstage/backstage#27147]
### 📜 Description

After upgrading from 1.31.3 to 1.32.0 I'm unable to open the scaffolder tasks page because it only shows with the database error below. Note my database being used is PostgreSQL.

### 👍 Expected behavior

Should open the tasks lists page.

### 👎 Actual Behavior with Screenshots

```
error Filtered internal error with logId=2fedba05cd0a982eb943 from response select count("tasks"."id") as "count" from "tasks" where "created_by" in ($1) order by "created_at" desc - column "tasks.created_at" must appear in the GROUP BY clause or be used in an aggregate function type=errorHandler logId=2fedba05cd0a982eb943 length=179 name=error severity=ERROR code=42803 detail=undefined hint=undefined position=88 internalPosition=undefined internalQuery=undefined where=undefined schema=undefined table=undefined column=undefined dataType=undefined constraint=undefined file=parse_agg.c line=1438 routine=check_ungrouped_columns_walker stack=error: select count("tasks"."id") as "count" from "tasks" where "created_by" in ($1) order by "created_at" desc - column "tasks.created_at" must appear in the GROUP BY clause or be used in an aggregate function
[backend]:     at Parser.parseErrorMessage (/workspaces/backstage/node_modules/pg-protocol/src/parser.ts:368:69)
[backend]:     at Parser.handlePacket (/workspaces/backstage/node_modules/pg-protocol/src/parser.ts:187:21)
[backend]:     at Parser.parse (/workspaces/backstage/node_modules/pg-protocol/src/parser.ts:102:30)
[backend]:     at Socket.<anonymous> (/workspaces/backstage/node_modules/pg-protocol/src/index.ts:7:48)
[backend]:     at Socket.emit (node:events:519:28)
[backend]:     at Socket.emit (node:domain:488:12)
[backend]:     at addChunk (node:internal/streams/readable:559:12)
[backend]:     at readableAddChunkPushByteMode (node:internal/streams/readable:510:3)
[backend]:     at Readable.push (node:internal/streams/readable:390:5)
[backend]:     at TCP.onStreamRead (node:internal/stream_base_commons:191:23)
```

### 👟 Reproduction steps

1. Use postgres db and open scaffolder tasks list page

### 🖥️ Your Environment

Backstage 1.32.0

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by SmajlovicA** [comment]
I have same issue after upgrading to 1.32.0

**Comment by stephenglass** [comment]
> I have same issue after upgrading to 1.32.0
> 
> If I try to use older image (container) that is version 1.31.3 then app is failing to start because of database error:
> 
> ForwardedError: Plugin 'catalog' startup failed; caused by Error: The migration directory is corrupt, the following files are missing: 20241003170511_alter_target_in_locations.js

unfortunately to rollback the update you will need to revert the knex database migration manually or if you are working in dev environment the fastest way is to drop the `backstage_plugin_catalog` database and restart the app

**Comment by stephenglass** [comment]
Fixed by #27159
